module/zfs/zfs_ioctl.c

```diff
@@ -2523,10 +2523,10 @@ zfs_prop_set_special(const char *dsname, zprop_source_t source,
 		err = zvol_set_volsize(dsname, intval);
 		break;
 	case ZFS_PROP_SNAPDEV:
-		err = zvol_set_snapdev(dsname, source, intval);
+		err = zvol_set_common(dsname, source, intval, ZFS_PROP_SNAPDEV);
```

Comment:
The order or arguments feels weird to me.  Property name should be before the value.  And may be before the source too.